### PR TITLE
refactor(compiler): remove direct accesses to DOM

### PR DESCRIPTION
**Pull request by marclaval**
Needed to run all compiler tests with a DOM adapter.

### CODE REVIEW

File: modules/angular2/src/core/compiler/url_resolver.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ export class UrlResolver {
 
   resolve(baseUrl: string, url: string): string {
     if (isBlank(baseUrl)) {
-      UrlResolver.a.href = url;
+      DOM.resolveAndSetHref(UrlResolver.a, url, null);
       return UrlResolver.a.href;
```
Comment (vicb) [inline modules/angular2/src/core/compiler/url_resolver.js]: `a.href` should probably go through the facade ?

File: modules/angular2/src/dom/browser_adapter.dart [hunk 2]
```diff
@@ -192,4 +192,26 @@ class BrowserDomAdapter extends DomAdapter {
   bool isStyleRule(CssRule rule) => rule is CssStyleRule;
   bool isMediaRule(CssRule rule) => rule is CssMediaRule;
   bool isKeyframesRule(CssRule rule) => rule is CssKeyframesRule;
+  void resolveAndSetHref(AnchorElement el, String baseUrl, String href) {
+    el.href = href == null? baseUrl: baseUrl + '/../' + href;
```
Comment (vicb) [inline modules/angular2/src/dom/browser_adapter.dart]: `null ?` (missing ws * 2)

File: modules/angular2/src/core/compiler/url_resolver.js (hunk 1 above)
Comment (mlaval) [inline modules/angular2/src/core/compiler/url_resolver.js]: Done

File: modules/angular2/src/dom/browser_adapter.dart (hunk 2 above)
Comment (mlaval) [inline modules/angular2/src/dom/browser_adapter.dart]: Done

File: modules/angular2/src/dom/browser_adapter.dart [hunk 3]
```diff
@@ -192,4 +192,29 @@ class BrowserDomAdapter extends DomAdapter {
   bool isStyleRule(CssRule rule) => rule is CssStyleRule;
   bool isMediaRule(CssRule rule) => rule is CssMediaRule;
   bool isKeyframesRule(CssRule rule) => rule is CssKeyframesRule;
+  String getHref(AnchorElement element) {
```
Comment (mhevery) [inline modules/angular2/src/dom/browser_adapter.dart]: I am kind of concerned that we have duplicated this code. Any way to put it in a shared space?

Comment (mlaval) [inline modules/angular2/src/dom/browser_adapter.dart]: Pushed this logic to an abstract `GenericBrowserDomAdapter` which both browser adapters extend.

**Pull request closed without merging**